src/Glennsl__BsBenchmarkJs.re

```diff
@@ -0,0 +1,971 @@
+type event;
+type options;
+type times;
+type stats;
+type platform;
+type support;
+type testFn = (. unit) => unit;
+type setupFn = (. unit) => unit;
+type teardownFn = (. unit) => unit;
+type eventHandler = (. event) => unit;
+type eventType = [ | `abort | `complete | `cycle | `error | `reset | `start | `unknown];
+
+let eventTypeFromString: string => eventType =
+  fun
+  | "abort" => `abort
+  | "complete" => `complete
+  | "cycle" => `cycle
+  | "error" => `error
+  | "reset" => `reset
+  | "start" => `start
+  | _ => `unknown;
+
+let eventTypeToString: eventType => string =
+  fun
+  | `abort => "abort"
+  | `complete => "complete"
+  | `cycle => "cycle"
+  | `error => "error"
+  | `reset => "reset"
+  | `start => "start"
+  | `unknown => "";
+
+module Benchmark = {
+  
+  type t = benchmark;
+  
+  module Internal = {
+
+    [@bs.module "benchmark"] [@bs.new] external make: (string, testFn) => t = "Benchmark";
+    [@bs.module "benchmark"] [@bs.new] external makeWithOptions: (string, testFn, options) => t = "Benchmark";
+
+    [@bs.send.pipe: t] external run: t = "run";
+    [@bs.send.pipe: t] external runWithOptions: options => t = "run";
+
+    [@bs.send.pipe: t] external clone: t => t = "clone";
+    [@bs.send.pipe: t] external cloneWithOptions: (t, options) => t = "clone";
+
+  };
+
+  let make: (~options: options=?, string, testFn) => t = (~options=?, name, fn) =>
+    switch (options) {
+    | (None) => Internal.make(name, fn)
+    | (Some(opt)) => Internal.makeWithOptions(name, fn, opt)
+    };
+
+  let run: (~options: options=?, t) => t = (~options=?, benchmark) =>
+    switch (options) {
+    | None => Internal.run(benchmark)
+    | Some(opt) => Internal.runWithOptions(opt, benchmark)
+    };
+
+  let clone: (~options: options=?, t, t) => t =
+    (~options=?, benchmark) =>
+      switch (options) {
+      | None => Internal.clone(benchmark)
+      | Some(opt) => Internal.cloneWithOptions(benchmark, opt)
+      };
+
+  [@bs.get] external getAborted: t => bool = "aborted";
+
+  [@bs.get] external getCompiled: t => testFn = "compiled";
+  
+  [@bs.get] external getCycles: t => int = "cycles";
+  
+  [@bs.get] external getCount: t => int = "count";
+  
+  [@bs.get] external getError: t => Js.Null_undefined.t(Js.Exn.t) = "error";
+  
+  [@bs.get] external getFn: t => testFn = "fn";
+  
+  [@bs.get] external getHz: t => float = "hz";
+  
+  [@bs.get] external getRunning: t => bool = "running";
+  
+  [@bs.get] external getSetup: t => setupFn = "setup";
+  
+  [@bs.get] external getStats: t => stats = "stats";
+  
+  [@bs.get] external getTeardown: t => teardownFn = "teardown";
+  
+  [@bs.get] external getTime: t => times = "time";
+  
+  [@bs.get] external getName: t => string = "name";
+  
+  [@bs.get] external getOptions: t => options = "options";
+      
+  [@bs.send.pipe: t] external abort: t = "abort";
+  
+  [@bs.send.pipe: t] external toString: string = "toString";
+  
+  [@bs.send.pipe: t] external cmp: t => int = "compare";
```

Comment:
Good point. I guess I was trying to avoid conflicting with `Pervasives.compare` in case of a module open, this module is very likely to be opened. It's probably better to avoid polymorphic compare anyway, though.

Reply:
It's usually not a good idea to open modules containing lots of generic functions, other than hyperlocally.

Reply:
Changed this to `compare`.